scripts/provisioning-terraform.sh

```diff
@@ -168,6 +168,17 @@ TERRAFORM_TFVARS_PATH="${TERRAFORM_ENVIRONMENT_DIR}/terraform.tfvars"
 
 # "${HOME}"/.config/gcloud/application_default_credentials.json is a well-known location for application-default credentials
 GOOGLE_APPLICATION_CREDENTIALS_PATH="/root/.config/gcloud/application_default_credentials.json"
+RUNTIME_SCRIPT_FOLDER="$(mktemp -d)"
+
+cat <<EOF >"${RUNTIME_SCRIPT_FOLDER}/run.sh"
+if ! gsutil list -p "${DEFAULT_PROJECT}" | grep "gs://tf-state-${DEFAULT_PROJECT}/" ; then
+  echo "Terraform backend storage does not exists, creating..."
+  gsutil mb -p ${DEFAULT_PROJECT} gs://tf-state-${DEFAULT_PROJECT}
```

Comment:
How do we remove this bucket when destroying everything? With a check in `provisioning-terraform.sh` if the user passes a `destroy` command, we remove this bucket after `terraform destroy` is done perhaps?

Reply:
Essentially I did not want to remove the bucket since I want it to be reused across multiple resource creation / deletion.   I've added a check to remove the bucket starting from [line 258](https://github.com/GoogleCloudPlatform/solutions-viai-edge-provisioning-configuration/pull/40/files#diff-f7871aba0c35160d7653778beab0dc5f37e9b1e7bb44e84b8f05dafa95147c53R258)